test/Tester/StorageFacet/Feature.Implementations/TableStorageFeature.cs

```diff
@@ -0,0 +1,61 @@
+﻿using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Extensions.DependencyInjection;
+using Orleans;
+using Orleans.Runtime;
+using Tester.StorageFacet.Abstractions;
+using Tester.StorageFacet.Infrastructure;
+
+namespace Tester.StorageFacet.Implementations
+{
+    public class TableStorageFeature<TState> : IStorageFeature<TState>, ILifecycleParticipant<GrainLifecyleStage>, IConfigurableStorageFeature
+    {
+        private IStorageFeatureConfig config;
+        private bool activateCalled;
+
+        public string Name => this.config.StateName;
+        public TState State { get; set; }
+
+        public Task Save()
+        {
+            return Task.CompletedTask;
+        }
+
+        public string GetExtendedInfo()
+        {
+            return $"Table:{this.Name}-ActivateCalled:{this.activateCalled}, StateType:{typeof(TState).Name}";
+        }
+
+        public Task LoadState(CancellationToken ct)
+        {
+            this.activateCalled = true;
+            return Task.CompletedTask;
+        }
+
+        public void Participate(ILifecycleObservable<GrainLifecyleStage> lifecycle)
+        {
+            lifecycle.Subscribe(GrainLifecyleStage.SetupState, LoadState);
+        }
+
+        public void Configure(IStorageFeatureConfig cfg)
+        {
+            this.config = cfg;
+        }
+    }
+
+    public class TableStorageFeatureFactory : StorageFeatureFactory
+    {
+        public TableStorageFeatureFactory(IGrainActivationContext context) : base(typeof(TableStorageFeature<>), context)
```

Comment:
> This is already a factory, so it makes sense that it is the responsibility of this derived class to decide how it instantiates.

Yes, it is the responsibility of this class how it instantiates the concrete type.  For convenience and reuse, I introduced a common base class, rather than duplicating code.  This does not alter there responsibility, nor prevent other factories, from doing this any way a developer wishes.  The Table and Blob implementation of the factories could have been entirely different, one newing an a impl that takes the config in it's constructor, the other deferring to the container and configuring after construction..?  This is just how I wrote it to avoid code duplication.

> Also, my preference would be that this derived class instead of just using the container to resolve, that it uses the new keyword or the ActivatorUtilities class, but pass the configuration to the constructor, instead of requiring a separate step to pass the configuration to the feature.

A second step is necessary for any instance that needs be wired up to the grain lifecycle, so in a common pattern, which again is not enforced, just a convenience so I didn't have to write/maintain as much code, nothing was gained by passing the config in the constructor.

This code is here to demonstrate the general pattern of feature, config, and factory interfaces.  Each implementer can implement as they please in a way that suits their needs, so any user with your sensibilities can freely follow them, and nothing in the way this is lain out prevents that..